sched/semaphore/sem_post.c

```diff
@@ -89,7 +89,11 @@ int nxsem_post(FAR sem_t *sem)
 
   /* Check the maximum allowable value */
 
-  DEBUGASSERT(sem_count < SEM_VALUE_MAX);
+  if (sem_count >= SEM_VALUE_MAX)
+    {
+      leave_critical_section(flags);
+      return -EOVERFLOW;
```

Comment:
This violates POSIX. `sem_post` can't set errno to `EOVERFLOW`.

Reply:
Freebsd also has similar description about EOVERFLOW
https://man.freebsd.org/cgi/man.cgi?query=sem_post&sektion=3

Reply:
While QNX does not have it :) http://www.qnx.com/developers/docs/7.0.0/index.html#com.qnx.doc.neutrino.lib_ref/topic/s/sem_post.html
Let's discuss, or maybe spin a vote in mailing list

Reply:
This is a regression issue, the following commit remove the overflow check:
https://github.com/apache/nuttx/pull/7372/files#diff-d574e66ae803b2d22c93c05fa66a481b3d9a9bdd4459fd16d5223ba3cebcf549L95

Some OS and libc implementations already have similar implementations.
musl:
https://github.com/bminor/musl/blob/master/src/thread/sem_post.c#L11
There is a risk of sem_count overflowing if we do not limit it, at present, sabre-6quad/netnsh_wb is broken by #7372, I think we need fix the overflow issue first.

Reply:
I am wondering that whether the implementation can just allow to return the error code specified by spec or have some flexibility to return some additional error to report the implementation detail. For example, most sem_t implementation has at least 32bit count which is very hard to hit the overflow issue, but NuttX use 16bit count and then hit this problem more frequenctly than others.

Reply:
We can try to get back this implementation specific error code since it was there in the past